mcp/tools/tools_coder.go

```diff
@@ -0,0 +1,351 @@
+package mcptools
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/mark3labs/mcp-go/mcp"
+	mcpserver "github.com/mark3labs/mcp-go/server"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/util/ptr"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/workspacesdk"
+)
+
+type handleCoderReportTaskArgs struct {
+	Summary string `json:"summary"`
+	Link    string `json:"link"`
+	Emoji   string `json:"emoji"`
+	Done    bool   `json:"done"`
+}
```

Comment:
To merge this initially, let's take this out.

I'd like for the top-level `mcp` package to just be tools like an MCP SDK, and less specific to the agent if that makes sense.